apps/src/code-studio/components/progress/StageLockDialog.jsx

```diff
@@ -0,0 +1,284 @@
+import React from 'react';
+import Radium from 'radium';
+import { connect } from 'react-redux';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
```

Comment:
That's good feedback too. This dialog is a little awkward because they're somewhat related but not directly. Like you wouldn't need to hit "Lock" AND "Save"